File: JUNE-2025/29-06-2025/p1498.swift
class Solution {
    func numSubseq(_ nums: [Int], _ target: Int) -> Int {
         let mod = Int(1e9 + 7)
    let n = nums.count
    var nums = nums.sorted()
    
    var pow2 = [Int](repeating: 1, count: n)
    for i in 1..<n {
        pow2[i] = (pow2[i - 1] * 2) % mod
    }

    var left = 0, right = n - 1
    var result = 0
    
    while left <= right {
        if nums[left] + nums[right] <= target {
            result = (result + pow2[right - left]) % mod
            left += 1
        } else {
            right -= 1
        }
    }
    
    return result
    }
}

File: JUNE-2025/29-06-2025/p1498.kt
class Solution {
    fun numSubseq(n: IntArray, t: Int): Int {
        val M = 1_000_000_007; n.sort(); var cnt = 0
        val f = IntArray(n.size); f[0] = 1; var j = n.size - 1
        for (i in 1..<n.size) f[i] = (2 * f[i - 1]) % M
        for ((i, x) in n.withIndex()) {
            while (j >= 0 && x + n[j] > t) j--
            if (j < i) break
            cnt = (cnt + f[j - i]) % M
        }
        return cnt
    }
}